test/relations.test.js

```diff
@@ -4987,7 +4987,7 @@ describe('relations', function() {
       Category.findById(category.id, function(err, cat) {
         if (err) return done(err);
         var link = cat.items.at(0);
-        link.updateAttributes({notes: 'Updated notes...'}, function(err, link) {
```

Comment:
I am confused that calling `connector.updateAttributes()` and callback the returned result is not the last step in `DAO.prototype.updateAttributes()`:
https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/dao.js#L3229
So in this test case, `link.updateAttributes()` still wants to execute the code in `updateAttributesCallback()`after L3229 **to do the real update**  after `connector.updateAttributes()` returns result, which is null.

This makes it difficult to return id not found error in `SQL.prototype.update()`  from loopback-connector/sql.js, because sometimes when `SQL.prototype.update()` get called, the caller expects error returned when `count===0`, while sometimes the caller tolerant it, and will do the real update afterwards.